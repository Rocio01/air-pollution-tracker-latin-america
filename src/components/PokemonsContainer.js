import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PokemonsHeader from './PokemonsHeader';
import { fetchPokemons } from '../redux/pokemons/pokemons';
import Pokemon from './Pokemon';

const PokemonsContainer = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchPokemons());
  }, []);
  const pokemonsState = useSelector((state) => state.pokemonsReducer.pokemons);
  return (
    <>

      <PokemonsHeader />

      <h6 className="m-2 details-title">DETAILS BY POKEMON</h6>
      <div className="row pokemon-container">
        {pokemonsState.map((pokemon) => (
          <Pokemon
            key={pokemon.name}
            name={pokemon.name}
            image={pokemon.sprites.front_shiny}
            base_experience={pokemon.base_experience}
            ability={pokemon.abilities[0].ability.name}
          />
        ))}

      </div>
    </>
  );
};

export default PokemonsContainer;
