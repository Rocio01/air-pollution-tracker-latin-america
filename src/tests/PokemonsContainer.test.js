import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import PokemonsContainer from '../components/PokemonsContainer';
import store from '../redux/configureStore';

describe('pokemons container component', () => {
  test('matches PokemonsContainer component snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <PokemonsContainer />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
