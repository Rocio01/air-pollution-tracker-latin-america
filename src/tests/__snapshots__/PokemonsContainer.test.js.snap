// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemons container component matches MissionsContainer component snapshot 1`] = `
<div
  className="pokemons-container"
>
  <div
    className="header row"
  >
    <img
      alt="pokemons header"
      className="pokemons-header col-6"
      src="Bulbasaur.PNG"
    />
    <div
      className="col-5 mt-5"
    >
      <h2
        className="title-header"
      >
        POKEMONS
      </h2>
      <p
        className="number-pokemons"
      >
        30 awesome pokemons
      </p>
    </div>
  </div>
  <h6
    className="m-2 details-title"
  >
    DETAILS BY POKEMON
  </h6>
  <div
    className="row pokemon-container"
  />
</div>
`;
